### PR TITLE
Added docs generation

### LINKED ISSUE #45: Write documentation
Currently, `opstooling-js` has a lot of utility methods that, unless you jump to read the code, are invisible. So it usually ends up with @mutantcornholio commenting in the code reviews that there is already a method for that. 

Let's write a clear and concise documentation showing a bit of each method and linking it to the file if necessary (or adding an example on how to use it). This would improve visibility and usability.

We can also use a doc generation tool using the documented methods.

### PR DESCRIPTION
**Pull request by Bullrich**
Closes #45 

Added automatic generation of docs using `typedoc`.

Also added docs to `validate` method.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,13 +5,17 @@ This repository hosts shared functionality for
 
 ![GitHub Issue Sync](https://github.com/paritytech/opstooling-js/actions/workflows/github-issue-sync.yml/badge.svg)
 
-# Installation
+## Docs
+
+![Docs deployed](https://github.com/paritytech/opstooling-js/actions/workflows/deploy-docs.yml/badge.svg)
 
-Choose a [tag](https://github.com/paritytech/opstooling-js/tags) and then run
+Find the docs in [paritytech.github.io/ostooling-js](https://paritytech.github.io/ostooling-js). They are deployed automatically.
```
Comment (mutantcornholio) [inline README.md]:
```suggestion
Find the docs in [paritytech.github.io/ostooling-js](https://paritytech.github.io/osptooling-js). They are deployed automatically.
```

**Review by mutantcornholio — COMMENTED** [review]
Do you want to provide docs for other methods as well?

**Comment by Bullrich** [comment]
> Do you want to provide docs for other methods as well?

The docs will be generated for all the methods available from index.ts

image

File: README.md (hunk 1 above)
Comment (Bullrich) [reply README.md]: Good catch but it seems we both misstyped.

I forgot the p and you put it in the wrong place 🤣 

Gonna fix that

Comment (mutantcornholio) [reply README.md]: ahaha, touché

**Review by rzadp — APPROVED** [review]

**Review by mutantcornholio — APPROVED** [review]

**Comment by rzadp** [comment]
Maybe we could use it in `js-integrations` as well.

**Pull request merged**